### PR TITLE
bluetooth: rpc: Modification of bt_conn_auth_cb_register function

### PR DESCRIPTION
**Pull request by maje-emb**
Serialization of the structure of callbacks when it is different
from NULL.

Signed-off-by: Marcin Jeliński <marcin.jelinski@nordicsemi.no>

### CODE REVIEW

**Review by KAGA164 — APPROVED** [review]
Looks good. Thanks

**Review by doki-nordic — CHANGES_REQUESTED** [review]
When on the application side user sets `NULL` parameter to `bt_conn_auth_cb_register`, it will not propagate to the network core.

Consider doing following changes: [bt_conn_auth_cb_register](https://github.com/maje-emb/sdk-nrf/pull/1/files)

**Comment by maje-emb** [comment]
> When on the application side user sets `NULL` parameter to `bt_conn_auth_cb_register`, it will not propagate to the network core.
> 
> Consider doing following changes: https://github.com/maje-emb/sdk-nrf/pull/1/files

It's a good point. Thanks.

File: subsys/bluetooth/rpc/client/bt_rpc_conn_client.c [hunk 1]
```diff
@@ -1588,17 +1588,21 @@ int bt_conn_auth_cb_register(const struct bt_conn_auth_cb *cb)
...
+		flags |= cb->passkey_display ? FLAG_PASSKEY_DISPLAY_PRESENT : 0;
+		flags |= cb->passkey_entry ? FLAG_PASSKEY_ENTRY_PRESENT : 0;
+		flags |= cb->passkey_confirm ? FLAG_PASSKEY_CONFIRM_PRESENT : 0;
+		flags |= cb->oob_data_request ? FLAG_OOB_DATA_REQUEST_PRESENT : 0;
+		flags |= cb->cancel ? FLAG_CANCEL_PRESENT : 0;
+		flags |= cb->pairing_confirm ? FLAG_PAIRING_CONFIRM_PRESENT : 0;
+		flags |= cb->pairing_complete ? FLAG_PAIRING_COMPLETE_PRESENT : 0;
+		flags |= cb->pairing_failed ? FLAG_PAIRING_FAILED_PRESENT : 0;
+	} else {
+		flags |= FLAG_AUTH_CB_IS_NULL;
```
[the remark this answers is not in the document]
Comment (maje-emb) [reply subsys/bluetooth/rpc/client/bt_rpc_conn_client.c]: Yes. It will be more clear. Fixed.

**Review by KAGA164 — APPROVED** [review]

File: subsys/bluetooth/rpc/host/bt_rpc_conn_host.c [hunk 2]
```diff
@@ -1287,6 +1287,12 @@ void bt_rpc_auth_cb_pairing_failed(struct bt_conn *conn, enum bt_security_err re
 
 static int bt_conn_auth_cb_register_on_remote(uint16_t flags)
 {
+	if (flags & FLAG_AUTH_CB_IS_NULL) {
+
```
Comment (grochu) [inline subsys/bluetooth/rpc/host/bt_rpc_conn_host.c:1291]: Blank lines not needed (this one and line 1293)

**Review by grochu — COMMENTED** [review]
One minor comment.

File: subsys/bluetooth/rpc/host/bt_rpc_conn_host.c (hunk 2 above)
Comment (maje-emb) [reply subsys/bluetooth/rpc/host/bt_rpc_conn_host.c]: Fixed.

**Review by grochu — APPROVED** [review]

**Review by doki-nordic — APPROVED** [review]

**Pull request merged**